diff -ruN -x '*.orig' llama-pristine/common/arg.cpp llama-sieos-src/common/arg.cpp
--- a/common/arg.cpp
+++ b/common/arg.cpp
@@ -47,6 +47,8 @@
 #   endif
 #elif defined(_AIX)
 #include <sys/limits.h>
+#elif defined(__sieos__)
+#include <limits.h>
 #else
 #include <sys/syslimits.h>
 #endif
diff -ruN -x '*.orig' llama-pristine/common/common.cpp llama-sieos-src/common/common.cpp
--- a/common/common.cpp
+++ b/common/common.cpp
@@ -1028,7 +1028,7 @@
     cache_directory = common_get_env("LLAMA_CACHE");
     if (cache_directory.empty()) {
 #if defined(__linux__) || defined(__FreeBSD__) || defined(_AIX) || \
-        defined(__OpenBSD__) || defined(__NetBSD__)
+        defined(__OpenBSD__) || defined(__NetBSD__) || defined(__sieos__)
         const std::string xdg_cache_home = common_get_env("XDG_CACHE_HOME");
         const std::string home           = common_get_env("HOME");
         if (!xdg_cache_home.empty()) {
@@ -1079,7 +1079,7 @@
         return p;
     };
 #if defined(__linux__) || defined(__FreeBSD__) || defined(_AIX) || \
-        defined(__OpenBSD__) || defined(__NetBSD__) || defined(__APPLE__)
+        defined(__OpenBSD__) || defined(__NetBSD__) || defined(__sieos__) || defined(__APPLE__)
     const std::string xdg_config_home = common_get_env("XDG_CONFIG_HOME");
     const std::string home            = common_get_env("HOME");
     if (!xdg_config_home.empty()) {
diff -ruN -x '*.orig' llama-pristine/common/download.cpp llama-sieos-src/common/download.cpp
--- a/common/download.cpp
+++ b/common/download.cpp
@@ -30,6 +30,8 @@
 #   endif
 #elif defined(_AIX)
 #include <sys/limits.h>
+#elif defined(__sieos__)
+#include <limits.h>
 #else
 #include <sys/syslimits.h>
 #endif
diff -ruN -x '*.orig' llama-pristine/src/llama-mmap.cpp llama-sieos-src/src/llama-mmap.cpp
--- a/src/llama-mmap.cpp
+++ b/src/llama-mmap.cpp
@@ -702,7 +702,7 @@
 
         char* errmsg = std::strerror(errno);
         bool suggest = (errno == ENOMEM);
-#if defined(TARGET_OS_VISION) || defined(TARGET_OS_TV) || defined(_AIX) || defined(__HAIKU__)
+#if defined(TARGET_OS_VISION) || defined(TARGET_OS_TV) || defined(_AIX) || defined(__HAIKU__) || defined(__sieos__)
         // visionOS/tvOS/Haiku don't support RLIMIT_MEMLOCK
         // Skip resource limit checks on these platforms
         suggest = false;
